std/algorithm/comparison.d

```diff
@@ -732,75 +731,130 @@ template equal(alias pred = "a == b")
     See_Also:
         $(HTTP sgi.com/tech/stl/_equal.html, STL's _equal)
     +/
-    bool equal(Range1, Range2)(Range1 r1, Range2 r2)
-    if (isInputRange!Range1 && isInputRange!Range2 &&
-        is(typeof(binaryFun!pred(r1.front, r2.front))))
+    bool equal(R, Ss...)(R r, Ss ss)
+    if (ss.length >= 1 &&
+        allSatisfy!(isInputRange, R, Ss) &&
+        (!isInfinite!R || !allSatisfy!(isInfinite, Ss)) // at least one must be finite
```

Comment:
It's not that interesting that one range is infinite - the answer will always be false. Better just disable the function for all infinite ranges.

Reply:
There are already unittests that checks results of infinite inputs and we don't want to regress right? Search for instance for `ReferenceInfiniteForwardRange`.